account_cutoff_base/models/account_cutoff.py

```diff
@@ -0,0 +1,391 @@
+        return list(self.cutoff_type_label_map.items())
+
+    @api.model
+    def _default_cutoff_account_id(self):
+        """Default account muast always be None"""
+        return None
+
+    @api.model
+    def _default_cutoff_journal_id(self):
+        return self.env.user.company_id.default_cutoff_journal_id
+
+    cutoff_date = fields.Date(
+        string='Cut-off Date', readonly=True,
+        states={'draft': [('readonly', False)]}, copy=False,
+        track_visibility='onchange'
+    )
+    cutoff_type = fields.Selection(
+        selection='_selection_cutoff_type',
+        string='Type',
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.context.get('cutoff_type'),
+        states={'draft': [('readonly', False)]},
+        oldname='type'
+    )
+    move_id = fields.Many2one(
+        'account.move',
+        string='Cut-off Journal Entry',
+        readonly=True,
+        copy=False
+    )
+    move_label = fields.Char(
+        string='Label of the Cut-off Journal Entry',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self._default_move_label(),
+        help="This label will be written in the 'Name' field of the "
+        "Cut-off Account Move Lines and in the 'Reference' field of "
+        "the Cut-off Account Move."
+    )
+    cutoff_account_id = fields.Many2one(
+        comodel_name='account.account',
+        string='Cut-off Account',
+        domain=[('deprecated', '=', False)],
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self._default_cutoff_account_id()
+    )
+    cutoff_journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string='Cut-off Account Journal',
+        default=lambda self: self._default_cutoff_journal_id(),
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    total_cutoff_amount = fields.Monetary(
+        compute='_compute_total_cutoff',
+        string="Total Cut-off Amount",
+        currency_field='company_currency_id',
+        readonly=True, track_visibility='onchange'
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        string='Company',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.cutoff')
+    )
+    company_currency_id = fields.Many2one(
+        related='company_id.currency_id',
+        readonly=True,
+        string='Company Currency'
+    )
+    line_ids = fields.One2many(
+        comodel_name='account.cutoff.line',
+        inverse_name='parent_id',
+        string='Cut-off Lines',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    state = fields.Selection(
+        selection=[('draft', 'Draft'), ('done', 'Done')],
+        string='State',
```

Comment:
The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
```